falcon/cyutil/misc.pyx

```diff
@@ -38,3 +38,14 @@ def isascii(unicode string not None):
             return False
 
     return True
+
+
+def encode_items_to_latin1(dict data not None):
```

Comment:
using cpdef has no advantage here, even if we cython compile response? Does it only has advantages when called from pyx files?

Reply:
Might be worth looking at this in the future, but currently the ASGI source tree is unfortunately not cythonized (although readied for that when we can; see also https://github.com/falconry/falcon/pull/1860#issue-571338714).

Reply:
I'm actually not sure if cpdef takes affect from cythonized py files or not

Reply:
It can not from pure Python (including cythonized Python), _methinks_.

It only works from inside a `cdef` -- you basically don't need to go via a full Python call, but can partially do it in C.

Reply:
For reference, I have so far stayed away from `cpdef`, and just wrote separate `cdef` and `def` functions, invoking the former from the latter where needed (in order to share code). I think it's basically what `cpdef` does.